pkg/operator/util/apply.go

```diff
@@ -21,3 +37,78 @@ func ApplyAsset(h *resource.Helper, assetPath string, hLog log.FieldLogger) erro
 	assetLog.Infof("asset applied successfully: %v", result)
 	return nil
 }
+
+// ApplyAssetWithGC loads a path from our bindata assets, adds an OwnerReference to the HiveConfig
+// for garbage collection (used when uninstalling Hive), and applies it to the cluster.
+func ApplyAssetWithGC(h *resource.Helper, assetPath string, hc *hivev1.HiveConfig, hLog log.FieldLogger) error {
+	assetLog := hLog.WithField("asset", assetPath)
+	assetLog.Info("reading asset")
+	runtimeObj, err := readRuntimeObject(assetPath)
+	if err != nil {
+		return err
+	}
+	assetLog.Info("applying asset with GC")
+	result, err := ApplyRuntimeObjectWithGC(h, runtimeObj, hc)
+	if err != nil {
+		assetLog.WithError(err).Error("error applying asset")
+		return err
+	}
+	assetLog.Infof("asset applied successfully: %v", result)
+	return nil
+}
+
+// ApplyAssetWithNSOverrideAndGC loads the given asset, overrides the namespace, adds an owner reference to
+// HiveConfig for uninstall, and applies it to the cluster.
+func ApplyAssetWithNSOverrideAndGC(h *resource.Helper, assetPath, namespaceOverride string, hiveConfig *hivev1.HiveConfig) error {
+	requiredObj, _, err := coreDeserializer.Decode(assets.MustAsset(assetPath), nil, nil)
+	if err != nil {
+		return errors.Wrapf(err, "unable to decode asset: %s", assetPath)
+	}
+	obj, _ := meta.Accessor(requiredObj)
+	obj.SetNamespace(namespaceOverride)
+	_, err = ApplyRuntimeObjectWithGC(h, requiredObj, hiveConfig)
+	if err != nil {
+		return errors.Wrapf(err, "unable to apply asset: %s", assetPath)
+	}
+	return nil
+}
+
+// ApplyClusterRoleBindingAssetWithSubjectNSOverrideAndGC loads the given asset, overrides the namespace on the subject,
+// adds an owner reference to HiveConfig for uninstall, and applies it to the cluster.
+func ApplyClusterRoleBindingAssetWithSubjectNSOverrideAndGC(h *resource.Helper, roleBindingAssetPath, namespaceOverride string, hiveConfig *hivev1.HiveConfig) error {
+
+	rb := resourceread.ReadClusterRoleBindingV1OrDie(assets.MustAsset(roleBindingAssetPath))
+	for i := range rb.Subjects {
+		if rb.Subjects[i].Kind == "ServiceAccount" || rb.Subjects[i].Namespace != "" {
+			rb.Subjects[i].Namespace = namespaceOverride
+		}
+	}
+	_, err := ApplyRuntimeObjectWithGC(h, rb, hiveConfig)
+	if err != nil {
+		return errors.Wrapf(err, "unable to apply asset: %s", roleBindingAssetPath)
+	}
+	return nil
+}
+
+// ApplyRuntimeObjectWithGC adds an OwnerReference to the HiveConfig on the runtime object, and applies it to the cluster.
+func ApplyRuntimeObjectWithGC(h *resource.Helper, runtimeObj runtime.Object, hc *hivev1.HiveConfig) (resource.ApplyResult, error) {
+	obj, err := meta.Accessor(runtimeObj)
+	if err != nil {
+		panic(err)
```

Comment:
Do we really want to panic here? Is the intention that we want hive-operator to crash-loop when there are bad assets?

Reply:
I added the error return after and forgot to replace the panic, will fix when I squash.